flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/enumerator/ContinuousIcebergEnumerator.java

```diff
@@ -161,7 +167,12 @@ private void processDiscoveredSplits(ContinuousEnumerationResult result, Throwab
         LOG.info("Update enumerator position to {}", result.toPosition());
       }
     } else {
-      LOG.error("Failed to discover new splits", error);
+      if (scanContext.maxAllowedPlanningFailures() == -1) {
```

Comment:
Here, we check the consecutive failures. if it is over the threshold, exception is throw. otherwise, just log an error and move on.

if it is a successful planning, the consecutive failure counter is reset to 0

Reply:
@stevenzwu You mean we don’t retry in `discoverSplits()`, but judge the number of errors here?

Reply:
@hililiwei that is correct. similar to Flink's tolerable checkpoint failures.

https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/config/#execution-checkpointing-tolerable-failed-checkpoints